src/App.tsx

```diff
@@ -1,14 +1,62 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { FilterTodoState } from './types/FilterTodoState';
+import { filterTodoByStatus, filterTodoByTitle } from './helpers/todoFilters';
+import { getFilteredBy } from './helpers/getFilteredBy';
+
+const initialFormState: FilterTodoState = {
+  todoStatus: TodoStatus.All,
+  searchQuery: '',
+};
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo []>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [filterState, setFilterState] = useState<FilterTodoState>(initialFormState);
+
+  const { searchQuery, todoStatus } = filterState;
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos().then(responce => {
+      setTodos(responce);
+      setLoading(false);
+    });
+  }, []);
+
+  const updateFormState = (key: string, value: string | TodoStatus) => {
+    setFilterState(prev => ({
+      ...prev,
+      [key]: value,
+    }));
+  };
+
+  const selectedTodo = todos
+    .find(todo => todo.id === selectedTodoId);
+
```

Comment:
Maybe we can set full todo in the state instead of only id, to not search it every time?